packages/design-system-scripts/gulp/docs/generatePages/generateDocPage.js

```diff
@@ -0,0 +1,90 @@
+const Docs = require('@cmsgov/design-system-docs/src/scripts/Docs').default;
+const React = require('react');
+const ReactDOMServer = require('react-dom/server');
+const savePage = require('./savePage');
+
+/**
+ * Create an HTML page with the documentation's UI
+ * @return {Promise}
+ */
+function generateDocPage(routes, page, docsPath, { rootPath, githubUrl }) {
+  const componentRenderer = () => {
+    if (process.env.NODE_ENV === 'development') {
+      // In development mode we let the client handle all of the React rendering,
+      // since if we were generating the HTML pages in our build process, Gulp would
+      // need restarted each time a React file changes, which is super annoying.
+      return '';
+    }
+
+    // On the client-side the "root" and "githubUrlBase" variables are defined
+    // via Webpack, but we also need to define them here for "server-side" rendering
+    process.env.root = rootPath;
+    process.env.githubUrlBase = githubUrl;
```

Comment:
Notes:
> Documentation pages are fed data through `window.pages` and `window.routes`. I want to make the docs package public and remove its dependencies to relative files outside its package folder, so I need to pass in necessary data at build time through global variables in the same fashion. For example, I want to pass in the github url root for the repository for which we're building a docs site. 

This line is used when we statically render the html files. The corresponding definition of this for the webpack dev server (`yarn start`) is in `packages/docs/webpack.config.js` using the define plugin.

Reply:
What do you think about eventually moving the webpack config into the scripts package? Is that possible?

Reply:
I was debating whether or not to move it into the scripts package. It is so tightly coupled to the build scripts that I think it might be a more appropriate place to have it there. But then I've also thought that the react components for the docs site are also tightly coupled to the build script, so if I were to move the webpack config, I may as well move the react components too and just keep the docs package as the documentation itself. I couldn't come to a solid decision on my own, so I went with having fewer changes for you to review.

Reply:
I could see the whole `docs/src/scripts` folder moving out of the docs package if we wanted that package to just be documentation and the scripts package deals with how to compile it all

Reply:
Ah I see what you mean, so you didnt want to move the `webpack.config` without also moving `docs/src/scripts`? Down the line I was also thinking of doing moving that into scripts so that child DS's could get updates to the doc site even if they only have documentation content. This way the experience of writing documentation would be something along the lines of [Docz's build in components](https://www.docz.site/docs/built-in-components).